reactive/kotlinx-coroutines-rx3/src/RxAwait.kt

```diff
@@ -41,9 +41,8 @@ public suspend fun CompletableSource.await(): Unit = suspendCancellableCoroutine
  * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this
  * function immediately resumes with [CancellationException] and disposes of its subscription.
  */
-@Suppress("UNCHECKED_CAST")
-public suspend fun <T> MaybeSource<T>.awaitSingleOrNull(): T? = suspendCancellableCoroutine { cont ->
-    subscribe(object : MaybeObserver<T> {
+public suspend fun <T> MaybeSource<T & Any>.awaitSingleOrNull(): T? = suspendCancellableCoroutine { cont ->
+    subscribe(object : MaybeObserver<T & Any> {
         override fun onSubscribe(d: Disposable) { cont.disposeOnCancellation(d) }
         override fun onComplete() { cont.resume(null) }
         override fun onSuccess(t: T) { cont.resume(t) }
```

Comment:
`T & Any`

Reply:
This bound seems to be properly propagated from the declaration site

Reply:
Since the object is implementing `MaybeObserver<T & Any>`, I think the method parameter needs to be `T & Any` to match.

Or you know what, actually, even if we were implementing plain `MaybeObserver<T>`, we'd probably still need the method parameter to be `T & Any`, since the interface declares its method as requiring a `@NonNull T`:

https://github.com/ReactiveX/RxJava/blob/8e53e0ee1637791923a81fdb9d17dfe8c1569250/src/main/java/io/reactivex/rxjava3/core/MaybeObserver.java#L76

I think that's why, with the current parameter type of `T`, the build fails with the flags I posted in the build script.

Reply:
Thanks for pointing it out. T & Any is required since Kotlin 1.8.0, it was an omission on the language site